packages/utils/src/color/Color.ts

```diff
@@ -0,0 +1,305 @@
+import { colord, extend } from 'colord';
+import namesPlugin from 'colord/plugins/names';
+
+import type {
+    AnyColor,
+    HslaColor,
+    HslColor,
+    HsvaColor,
+    HsvColor,
+    RgbaColor,
+    RgbColor,
+} from 'colord/types';
+
+extend([namesPlugin]);
+
+/** Different possible value types for Color class */
+export type ColorSource = string | number | number[] | Float32Array
+| HslColor | HslaColor | HsvColor | HsvaColor | RgbColor | RgbaColor;
+
+/**
+ * Converts a color value to an [R,G,B,A] array of normalized values (numbers from 0.0 to 1.0).
+ * @example
+ * import { utils } from 'pixi.js';
+ * new utils.Color('red').toArray(); // [1, 0, 0, 1]
+ * new utils.Color(0xff0000).toArray(); // [1, 0, 0, 1]
+ * new utils.Color('ff0000').toArray(); // [1, 0, 0, 1]
+ * new utils.Color('#f00').toArray(); // [1, 0, 0, 1]
+ * new utils.Color([255, 0, 0, 0.5]).toArray(); // [1, 0, 0, 0.5]
+ * new utils.Color('rgb(255, 0, 0, 0.5)').toArray(); // [1, 0, 0, 0.5]
+ * new utils.Color({h: 0, s: 100, l: 50, a: 0.5}).toArray(); // [1, 0, 0, 0.5]
+ * new utils.Color({h: 0, s: 100, v: 100, a: 0.5}).toArray(); // [1, 0, 0, 0.5]
+ * @memberof PIXI.utils
+ */
+export class Color
+{
+    /**
+     * Default Color object for static uses
+     * @example
+     * import { utils } from 'pixi.js';
+     * utils.Color.default.setValue(0xffffff).toString(); // '#ffffff'
+     */
+    static readonly default = new Color();
+
+    /** Pattern for hex strings */
+    static readonly HEX_PATTERN = /^#?([a-f0-9]{3}){1,2}([a-f0-9]{2})?$/i;
+
+    /** Internal color source, from constructor or set value */
+    private _value: ColorSource;
+
+    /** Normalized rgba component, floats from 0-1 */
+    private _components: Float32Array;
+
+    /** Cache color as number */
+    private _int: number;
+
+    /**
+     * @param {PIXI.ColorSource} value - Optional value to use, if not provided, white is used.
+     */
+    constructor(value: ColorSource = 0xffffff)
+    {
+        this._components = new Float32Array(4);
+        this._components.fill(1);
+        this.value = value;
+    }
+
+    /**
+     * Set the value, suitable for chaining
+     * @param value
+     */
+    setValue(value: ColorSource): this
+    {
+        this.value = value;
+
+        return this;
+    }
+
+    /**
+     * Set the current color source
+     * @type {PIXI.ColorSource}
+     */
+    set value(value: ColorSource)
+    {
+        if (this._value !== value)
+        {
+            this.normalize();
+            this._value = value;
```

Comment:
`this.normalize()` need `this._value` as input :( You may make `value` as an argument of `normalize()`, or just merge `normalize()` into `set value` since it is only used here.

Reply:
Yes, I made it an argument instead.